generator/java/lib/Messages/MAVLinkPayload.java

```diff
@@ -72,7 +77,7 @@ public short getShort() {
         return result;
     }
 
-    public int getUnsignedShort(){
+    public int getUnsignedShort() {
```

Comment:
I think these functions does not handle truncated values.
payload.get(index) will give exception if index is out of range.

Reply:
The payload buffer is sized to the largest possible payload so those functions will just return 0.

Reply:
This is what I did to deal with truncated payloads:
https://github.com/ArduPilot/pymavlink/blob/4935e9cf039050093467e8b2670c0fb1b836e650/generator/java/lib/Messages/MAVLinkPayload.java#L34

Reply:
oic you preallocated with full size so it will never get out of index.